tensorflow/compiler/tf2tensorrt/convert/ops/tile.cc

```diff
@@ -0,0 +1,169 @@
+                                       nb_dims, ")");
+      }
+
+      bool positiveReplic = true;
+      for (int i = 0; i < nb_dims; i++)
+        positiveReplic &= multiplies[i] > 0;
+
+      if (!positiveReplic) {
+        char buffer[256], *pBuff = buffer;
+        for (int i = 0; i < nb_dims; i++)
+          pBuff += sprintf(pBuff, (i? ", %d" : "%d"), multiplies[i]);
+
+         return errors::InvalidArgument("All replications of the Tile operation in ", params.node_def.name(),
+                                     " should be positive, got (", buffer, ").");
+      }
+    }
+
+    return Status::OK();
+  }
+
+  Status Convert() {
+    const auto &params = *this->params_;
+    const auto& inputs = params.inputs;
+    auto *converter = params.converter;
+    auto *network = converter->network();
+    const auto &tensor = inputs.at(0);
+    const auto &replics = inputs.at(1);
+    const auto dims = tensor.GetTrtDims();
+    const auto nb_dims = dims.nbDims;
+    const auto dim_adj = params.use_implicit_batch && tensor.is_tensor()? 1 : 0;
+
+    bool dynamic_flag = false;
+    const int *pMultiplies = nullptr;
+    if (replics.is_tensor()) {
+      dynamic_flag = true;
+    } else {
+      pMultiplies = replics.weights().GetPointer<int>();
+    }
+
+    nvinfer1::Dims size{nb_dims, {1}};
+    nvinfer1::Dims stride{nb_dims, {1}};
+    const auto *pSize = dims.d;
+    for (int i = 1 - dim_adj; i < nb_dims; i++) {
+      stride.d[i] = 1;
+      if (pMultiplies)
+        dynamic_flag |= (size.d[i] = pMultiplies[dim_adj+i] * *(pSize+i)) < 0;
+    }
+
+    auto input_tensor = tensor.is_tensor()?
+                        tensor.tensor() :
+                        converter->CreateConstantLayer(tensor.weights(), dims);
+
+    auto &input_trt_tensor = *input_tensor->trt_tensor();
+    nvinfer1::ITensor* target_shape = nullptr;
+    if (dynamic_flag) {
+      nvinfer1::ITensor *shape = network->addShape(input_trt_tensor)->getOutput(0);
+      const ITensorProxyPtr multiplies = replics.is_tensor()?
+                                         replics.tensor()->trt_tensor() :
+                                         converter->CreateConstantLayer(replics.weights(), replics.GetTrtDims());
+      target_shape = network->addElementWise(*shape, *multiplies->trt_tensor(), nvinfer1::ElementWiseOperation::kPROD)
+                            ->getOutput(0);
+    }
+
+    const auto start = nvinfer1::Dims{nb_dims, {}};
+    auto layer = network->addSlice(input_trt_tensor, start, size, stride);
+    layer->setMode(nvinfer1::SliceMode::kWRAP);
+    if (target_shape)
+      layer->setInput(2, *target_shape);
+
+    converter->SetLayerName(layer, params.node_def.name(), "to_tile");
+    ITensorProxyPtr output_tensor = layer->getOutput(0);
+    if (tensor.is_weights() && params.use_implicit_batch) {
+      // Reshape output tensor by removing first dimension
+      const auto out_dims = output_tensor->getDimensions();
```

Comment:
You can use DimsAdapter and DimsAdapter::RemoveBatchDimension

Reply:
Sorry, I cannot find DimsAdapter in current version of `tf2tensorrt/convert/utils.h`. You seem to be mentioning brand new code that hasn't been merged yet.

Reply:
Implemented